lib/parser.js

```diff
@@ -26,20 +26,22 @@ module.exports = {
   registerSchemaParser,
   ParserError,
   AsyncAPIDocument,
+  parseIntents,
+  IntentAsyncAPIDocument
 };
 
 /**
- * Parses and validate an AsyncAPI document from YAML or JSON.
- * 
+ * Parse input to JSON object
+ * @private
  * @param {String} asyncapiYAMLorJSON An AsyncAPI document in JSON or YAML format.
  * @param {Object} [options] Configuration options.
  * @param {String} [options.path] Path to the AsyncAPI document. It will be used to resolve relative references. Defaults to current working dir.
  * @param {Object} [options.parse] Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
  * @param {Object} [options.resolve] Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
  * @param {Object} [options.applyTraits=true] Whether to resolve and apply traits or not.
- * @returns {Promise<AsyncAPIDocument>} The parsed AsyncAPI document.
+ * @returns {Promise<any>} The parsed AsyncAPI document.
  */
-async function parse(asyncapiYAMLorJSON, options = {}) {
+async function parseToJson(asyncapiYAMLorJSON, options) {
```

Comment:
Is this a BC within the current implementation? Is this method the one users are calling right now?

Reply:
This is instead of having duplicate functionality so `parseIntents` and `parse` shares the same functionality.

Reply:
Oh I see https://github.com/asyncapi/parser-js/pull/270/files#diff-2f45270c7389933f71b380538ff52e106ee5cb66a0b573d9967467dfdbc11c02R144. nvm then 👍